.buildkite/scripts/e2e/bootstrap.py

```diff
@@ -0,0 +1,150 @@
+"""
+E2E bootstrapping with Python script
+    - A script to resolve stack versions
+    - Download and spin up elastic-package
+    - Clone integrations repo and prepare packages
+    - When E2E finishes, teardown the stack
+"""
+import os
+import time
+from util import Util
+
+
+class Bootstrap:
+    ELASTIC_PACKAGE_DISTRO_URL = "https://api.github.com/repos/elastic/elastic-package/releases/latest"
+    LOGSTASH_CONTAINER_NAME = "elastic-package-stack-e2e-logstash-1"
+    PLUGIN_NAME = "logstash-filter-elastic_integration"
+
+    def __init__(self, stack_version: str, platform: str) -> None:
+        f"""
+        A constructor of the {Bootstrap}.
+
+        Args:
+            stack_version: An Elastic stack version where {Bootstrap} spins up with
+            platform: pass macos if you are on your dev Mac, otherwise linux 
+
+        Returns:
+            Validates platform, sets stack version and platform
+        """
+        self.stack_version = stack_version
+        self.__validate_platform(platform)
+        self.__set_distro(platform)
+
+    def __validate_platform(self, platform: str) -> None:
+        platforms = ["macos", "linux"]
+        if platform not in platforms:
+            raise ValueError(f"platform accepts {platforms} only")
+        self.platform = platform
+
+    def __set_distro(self, platform) -> None:
+        self.distro = "darwin_amd64.tar.gz" if platform == "macos" else "linux_amd64.tar.gz"
+
+    def __download_elastic_package(self) -> None:
+        response = Util.call_url_with_retry(self.ELASTIC_PACKAGE_DISTRO_URL)
+        release_info = response.json()
+
+        download_urls = [asset["browser_download_url"] for asset in release_info["assets"]]
+        download_url = [url for url in download_urls if self.distro in url][0]
+        if download_url.strip() == '':
+            raise Exception(f"Could not resolve elastic-package distro download URL, release info: {release_info}.")
+
+        file_name = "downloaded_elastic_package_" + self.distro
+        # downloading with `urllib3` gives a different size file which causes a corrupted file issue
+        Util.run_or_raise_error(["curl", "-o", file_name, "--retry", "5", "--retry-delay", "5", "-fSL", download_url],
+                                "Failed to download elastic-package")
+        print("elastic-package is successfully downloaded.")
+
+        # Extract the downloaded tar.gz file
+        Util.run_or_raise_error(["tar", "zxf", file_name],
+                                f"Error occurred while unzipping {file_name}")
+
+    def __make_elastic_package_global(self) -> None:
+        Util.run_or_raise_error(["sudo", "mv", "elastic-package", "/usr/local/bin"],
+                                "Could not make `elastic-package` global")
+
+    def __clone_integrations_repo(self) -> None:
+        Util.run_or_raise_error(["retry", "-t", "5", "--", "git", "clone", "--single-branch",
```

Comment:
Since I can't verify on my local, I wonder if there is any issue running `retry` on `subprocess`, @dliappis

Reply:
You can try it locally too but installing `retry`; if you are on macOS there is a port: https://formulae.brew.sh/formula/retry

Reply:
By the way this fails if the repo already exists (e.g. when testing it locally a few times):

```
fatal: destination path 'integrations' already exists and is not an empty directory.
retry: git returned 128, backing off for 10 seconds and trying again...
```

might be good to make it more resilient (but it will likely include a complex invocation of git commands and/or using the shell too)

Reply:
The expectation is to test with _recent_ integrations, I think continuing the test if repo exists would be an ideal here. On the other hand, it is a meaningless retry.

Reply:
Yes, if you want to implement this, you shouldn't rely on the native `retry` binary, but instead e.g. use a different helper function that re-runs git clone only if stderr doesn't include `fatal: destination path 'integrations' already exists and is not an empty directory.`.

Reply:
Couldn't come up with a better solution, if I use git directly losing retry or if I use retry. I have changed retry to 3 to find a baseline for now and if it causes frequent issue, checking existence of `integrations` folder and removing seems a better one for me.

Reply:
Yup, added to the Readme!